torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1405,7 +1407,7 @@ def solve(self, disable_congruences=True, disable_equivalences=True):
                         self._dynamic_results.add(self._dcp.doprint(arg))
                 else:
                     self._dynamic_results.add(self._dcp.doprint(solution))
-            except NotImplementedError as e:
+            except (NotImplementedError, AssertionError) as e:
```

Comment:
meh, not a huge fan of catching all assertion errors (you're liable to catch legit assertion problems) but I guess it's fine

Reply:
I agree too that "AssertionError" is semantically wrong here.   
Especially as next to the intended assertion there is another assertion that really is an assertion, i.e. something that should not fail.